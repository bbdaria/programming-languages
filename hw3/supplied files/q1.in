print "===TEST START===\n";
(* Testing to_binary function *)
to_binary 0;
to_binary 5;  
to_binary 1;  
to_binary 16;

(* Testing encode function *)
encode [1, 1, 0, 1];
encode [1, 0, 1, 1, 1, 1];
encode [0, 0, 0, 1];
encode [0, 1, 0, 0, 0, 0];
encode [1, 0, 0, 0, 0, 0];
encode [1, 1];

(* Testing decode function *)
decode ([0,1,0,1,1], 4);
decode ([0,1,0,0,1,1,0,0,1], 6);
decode ([1,0,0,1,1], 4);
decode ([1,0,1,1,0,0,0,0,1], 6);
decode ([0,1,1,1,0,0,0,0,1], 6);
decode ([0,1,1], 2);
