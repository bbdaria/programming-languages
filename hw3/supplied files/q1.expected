val it = () : unit
val it = [] : int list
val it = [1,0,1] : int list
val it = [1] : int list
val it = [0,0,0,0,1] : int list
val it = [0,1,0,1,1] : int list
val it = [0,1,0,0,1,1,0,0,1] : int list
val it = [1,0,0,1,1] : int list
val it = [1,0,1,1,0,0,0,0,1] : int list
val it = [0,1,1,1,0,0,0,0,1] : int list
val it = [0,1,1] : int list
val it = [1,1,0,1] : int list
val it = [1,0,1,1,1,1] : int list
val it = [0,0,0,1] : int list
val it = [0,1,0,0,0,0] : int list
val it = [1,0,0,0,0,0] : int list
val it = [1,1] : int list
- 
